### PR TITLE
 openshift/cluster-openshift-apiserver-operator: Add vendor check

### PR DESCRIPTION
**Pull request by tnozicka**
/cc @deads2k @sttts

### CODE REVIEW

File: ci-operator/config/openshift/cluster-openshift-apiserver-operator/openshift-cluster-openshift-apiserver-operator-master.yaml [hunk 1]
```diff
@@ -37,14 +37,23 @@ tests:
...
+    go get -u github.com/Masterminds/glide
+    make verify-deps -k --warn-undefined-variables
+  container:
+    from: src
+
 - as: unit
   commands: |
     go get -u github.com/openshift/release/tools/gotest2junit
-    make test-unit JUNITFILE=/tmp/artifacts/unit_report.xml
+    mkdir /tmp/artifacts
```
Comment (tnozicka) [inline ci-operator/config/openshift/cluster-openshift-apiserver-operator/openshift-cluster-openshift-apiserver-operator-master.yaml]: the current version semes to be not creating the dir anymore, adding the mkdir back

File: ci-operator/config/openshift/cluster-openshift-apiserver-operator/openshift-cluster-openshift-apiserver-operator-master.yaml [hunk 2]
```diff
@@ -37,14 +37,23 @@ tests:
...
   commands: |
     go get -u golang.org/x/tools/imports
-    make verify -k
+    make verify -k --warn-undefined-variables
   container:
     from: bin
 
+- as: verify-deps
+  commands: |
+    HOME=$( mktemp -d )
```
Comment (deads2k) [inline ci-operator/config/openshift/cluster-openshift-apiserver-operator/openshift-cluster-openshift-apiserver-operator-master.yaml]: why reassign home?

File: ci-operator/config/openshift/cluster-openshift-apiserver-operator/openshift-cluster-openshift-apiserver-operator-master.yaml [hunk 3]
```diff
@@ -37,14 +37,23 @@ tests:
...
+    go get -u github.com/Masterminds/glide
```
Comment (deads2k) [inline ci-operator/config/openshift/cluster-openshift-apiserver-operator/openshift-cluster-openshift-apiserver-operator-master.yaml]: this is not a path to happiness.  it is going to limit adoption if we have these fix ups lying around.

File: ci-operator/config/openshift/cluster-openshift-apiserver-operator/openshift-cluster-openshift-apiserver-operator-master.yaml (hunk 2 above)
Comment (tnozicka) [reply ci-operator/config/openshift/cluster-openshift-apiserver-operator/openshift-cluster-openshift-apiserver-operator-master.yaml]: this is set to a dir (I think `/`) which the user doesn't have permissions to, bad image

fyi @stevekuznetsov

File: ci-operator/config/openshift/cluster-openshift-apiserver-operator/openshift-cluster-openshift-apiserver-operator-master.yaml (hunk 3 above)
Comment (tnozicka) [reply ci-operator/config/openshift/cluster-openshift-apiserver-operator/openshift-cluster-openshift-apiserver-operator-master.yaml]: I think this was meant for HOME line, and I agree, just wanted to make it work without CI - temporary workaround

Comment (tnozicka) [reply ci-operator/config/openshift/cluster-openshift-apiserver-operator/openshift-cluster-openshift-apiserver-operator-master.yaml]: @deads2k Issue filled, we can move with the workaround in the meantime
https://github.com/openshift/release/issues/2057

File: ci-operator/config/openshift/cluster-openshift-apiserver-operator/openshift-cluster-openshift-apiserver-operator-master.yaml (hunk 2 above)
Comment (stevekuznetsov) [reply ci-operator/config/openshift/cluster-openshift-apiserver-operator/openshift-cluster-openshift-apiserver-operator-master.yaml]: `$HOME` is unset ... why do you need it?

**Pull request merged**